File: ViewModelFragment/app/src/main/java/com/example/solar/viewmodelfragment/fragment/fm2.java
package com.example.solar.viewmodelfragment.fragment;


import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.solar.viewmodelfragment.R;
import com.example.solar.viewmodelfragment.VMD;

public class fm2 extends Fragment {
    VMD vmd;
    TextView txtName;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_fm2, container, false);
     txtName=view.findViewById(R.id.txtName);
        vmd=ViewModelProviders.of(getActivity()).get(VMD.class);
        vmd.laydulieu().observe(this, new Observer<Integer>() {
            @Override
            public void onChanged(@Nullable Integer integer) {
                txtName.setText("Gia tri "+integer);
            }
        });

        return view;
    }

}

File: viewpager buttomNavigation diffUtil/app/src/main/java/com/example/myapplication/fragment/Fm2.kt
package com.example.myapplication.fragment


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.myapplication.Content
import com.example.myapplication.ContentAdapter

import com.example.myapplication.R
import kotlinx.android.synthetic.main.fragment_fm1.*
import kotlinx.android.synthetic.main.fragment_fm1.view.*
import kotlinx.android.synthetic.main.fragment_fm2.*

class Fm2 : Fragment() {
    private lateinit var contentAdapter: ContentAdapter
    private lateinit var listData: ArrayList<Content>
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_fm2, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        listData = ArrayList()
        contentAdapter = ContentAdapter(addListData())
        recylcerviewContent2.apply {
            adapter = contentAdapter
            layoutManager = LinearLayoutManager(context)
            setHasFixedSize(true)
        }
    }

    private fun addListData(): ArrayList<Content>? {

        listData.add(Content("Kjdksjdkfs", R.drawable.a))
        listData.add(Content("ddd", R.drawable.b))
        listData.add(Content("eee", R.drawable.c))
        listData.add(Content("ffff", R.drawable.d))
        listData.add(Content("yyt", R.drawable.e))
        listData.add(Content("jghjg", R.drawable.f))
        listData.add(Content("kuyu", R.drawable.g))
        listData.add(Content("rtyrtt", R.drawable.h))
        listData.add(Content("gfghgg", R.drawable.i))
        listData.add(Content("nnnnn", R.drawable.c))
        listData.add(Content("tttrr", R.drawable.d))
        listData.add(Content("ggjjj", R.drawable.f))
        listData.add(Content("mmmm", R.drawable.g))
        listData.add(Content("Kjdksjdkfs", R.drawable.a))
        listData.add(Content("ddd", R.drawable.b))
        listData.add(Content("eee", R.drawable.c))
        listData.add(Content("ffff", R.drawable.d))
        listData.add(Content("yyt", R.drawable.e))
        listData.add(Content("jghjg", R.drawable.f))
        listData.add(Content("kuyu", R.drawable.g))
        listData.add(Content("rtyrtt", R.drawable.h))

        return listData
    }
}
